### DISCUSSION TITLE
How to write data correctly?

### DISCUSSION THREAD

**Discussion by profoundmentalretardation** [pschatzmann/ESP32-A2DP#440] [Q&A] [answered]
Hello, I'm new to microcontroller programming and I have a question. How do I need to transfer data for recording via bluetooth correctly? I have this situation: I get audio data on trrs breakout, but I can't write it to the column using your library. Could you help me solve this problem? I get stereo by trrs breakout.

**Comment by pschatzmann** [comment, marked as answer]
Your description is slightly confusing and I am not sure if I understand you correctly: I assume you want to use the Bluetooth Sink and output the audio to your trrs breakout. This is simply what's already described in the Readme 'Output to the Internal DAC' chapter.

However I do not recommend to use the internal DAC for Audio but to use a proper external DAC as described in the Wiki.

**Reply by profoundmentalretardation** [reply]
No, I have Music Station (Yandex Station if you know). I'd like to use the station with soundbar, but the station cannot play audio via Bluetooth. The station has jack 3.5mm output. I made a thing that can listen to station's stream via aux and directs the stream via bluetooth to the soundbar. I cannot understand how to send stereo data to the soundbar using write_data function in BluetoothA2DPSource. I know how to work with BluetoothA2DPSink (thank you for your docs), but don't know how to work with write_data properly

**Comment by pschatzmann** [comment]
It is quite challenging to use A2DPSource, so if you are a beginner, it is nothing for you: Additional hint you need a proper Audio ADC that can provide stereo 16 bit at 44100 samples per second!

Thought I never used this, I would expect it is far easier to just use a [A2DP Module](https://de.aliexpress.com/item/33058710334.html?spm=a2g0o.order_list.order_list_main.4.21ef5c5fYlPNeU&gatewayAdapt=glo2deu)

**Reply by profoundmentalretardation** [reply]
Thank you! I'll try to do something)

**Reply by pschatzmann** [reply]
The other simple solution would be to use e.g. the AI Thinker AudioKit of a LyraT board which has a proper audio chip with a DAC and ADC together with an ESP32.  
The related sketch is incredibly simple: https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/basic-audiokit-a2dp/basic-audiokit-a2dp.ino

**Reply by profoundmentalretardation** [reply]
I bought this board. As far as I understand, I will just need to plug the aux cable into AUX_IN and copy the code from the example, changing the speaker name, right?
